dap4/d4tests/src/test/java/dap4/test/TestDAP4Client.java

```diff
@@ -0,0 +1,406 @@
+package dap4.test;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ucar.httpservices.HTTPException;
+import ucar.httpservices.HTTPFactory;
+import ucar.httpservices.HTTPMethod;
+import ucar.nc2.dataset.NetcdfDataset;
+import ucar.unidata.util.test.UnitTestCommon;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.lang.invoke.MethodHandles;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Test OpenDap Server at the NetcdfDataset level
+ */
+public class TestDAP4Client extends DapTestCommon {
+  private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  static final boolean DEBUG = false;
+
+  static final boolean BROKEN = false; // on/off known broken tests
+
```

Comment:
We can use annotations to turn off known broken tests, we shouldn't need a flag.

Reply:
Do you object to overloading the method name or should I use
a different name?

Reply:
This will be fixed in a later PR addressing major changes to DAP4